src/catalog/init.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+function snagshout_register_catalogs() {
+  register_post_type(
+    'ss_catalog',
+    [
+      'labels' => [
+        'name' => __('Catalogs', 'textdomain'),
+        'singular_name' => __('Catalog', 'textdomain'),
+        'add_new' => __('Add New', 'textdomain'),
+        'add_new_item' => __('Add New Catalog', 'textdomain'),
+        'new_item' => __('New Catalog', 'textdomain'),
+        'edit_item' => __('Edit Catalog', 'textdomain'),
+        'view_item' => __('View Catalog', 'textdomain'),
+        'all_items' => __('All Catalogs', 'textdomain'),
+        'search_items' => __('Search Catalogs', 'textdomain'),
+        'not_found' => __('No catalogs found.', 'textdomain'),
+        'not_found_in_trash' => __('No catalogs found in Trash.', 'textdomain'),
+      ],
+      'description' => 'Displays Snagshout deals',
+      'public' => true,
+      'hierarchical' => false,
+      'exclude_from_search' => true,
+      'menu_icon' => 'dashicons-cart',
+      'register_meta_box_cb' => 'snagshout_register_catalog_metabox',
+      'supports' => ['title'],
+      // ...
+    ]
+  );
+}
+
+function snagshout_register_catalog_metabox() {
+  add_meta_box(
+    'ss-catalog-options',
+    __('Catalog Options', 'textdomain'),
+    'snagshout_render_catalog_metabox',
+    'ss_catalog',
+    'side',
+    'high'
+  );
+}
+
+function snagshout_render_catalog_metabox($post) {
+  $categories = ['all' => 'All categories'];
+  $categoriesResponse = json_decode(snagshout_fetch_categories());
+
+  if ($categoriesResponse && $categoriesResponse->data) {
+    foreach ($categoriesResponse->data as $category) {
+      $categories[$category->id] = $category->name;
+    }
+  }
+
+  $instance = [];
+
+  echo snagshout_render_view('catalog-options', [
+    // General data.
+    'categories' => $categories,
+    'layouts' => [
```

Comment:
todo: DRY this up! Abstract these into constants so the widget and post type can share these definitions.